tests/test_with_decorators.py

```diff
@@ -7,58 +7,73 @@
 #   * https://docs.pytest.org/en/latest/example/parametrize.html#paramexamples
 
 
-# ~~~~ test for equality ~~~~
-# @pytest.mark.skip(reason="test function not implemented yet")
+# ~~~~ skip a test ~~~~
+@pytest.mark.skip(reason="test not fully implemented/functional yet")
+def test__this_is_an_extensive_test_that_is_not_fully_implemented_yet():
+    assert add(3, 2) == 5
+
+
+# ~~~~ mark a test as failing ~~~~
+@pytest.mark.xfail(reason="conversion of string arguments into int/float " +
+                          "not implemented yet")
+def test__this_test_will_fail_and_we_know_it():
+    assert add(3, "2") == 5
+
+
+# ~~~~ mark a test as failing but it does not fail ~~~~
```

Comment:
is this indicated as `xpassed`?

Reply:
The test is evaluated but its result fail/pass will nearly be ignored by pytest. In contrast, a test marked by `skip` will be completly ignored. Although a failing test marked by `xfail` won't throw an error/assertion, python does log whether the tests fails or passes. The pytest textual output shows a "x" for a failing test marked by `xfail` or "X" for a passing test marked by `xfail`. I wanted to present this difference in the output of pytest.

I think a reasonable use case for this is as follows:

* we create a test to test a planed feature of a function
* the function should provide this specific feature in future but does not work properly yet for some unknown reason
* instead of `skip`ping the test we set it to `xfail`
* although pytest won't break when the tests fails, we will see when the test passes at one point